### PR TITLE
Add default exceptions affected by suppress

### PR DESCRIPTION
**Pull request by ixti**
Allow use `suppress` without list of exceptions.
So that instead of `suppress(StandardError) { ... }` one will be able to simply call `suppress { ... }`
This makes `suppress` more inline with ruby's `rescue`.

```
suppress { do_something_that_might_fail }

# instead of

begin
  do_something_that_might_fail
rescue
end

# or

do_something_that_might_fail rescue nil
```

### CODE REVIEW

**Comment by nateberkopec** [comment]
Should we really be providing more convenient ways for people to `rescue nil`?

`suppress` is a very, very heavy tool, and should be used with caution and forethought. Allowing `suppress <block>` just allows difficult-to-debug failure-prone code IMO.

**Comment by rafaelfranca** [comment]
There is nothing that you can't do with `rescue` here. Or either with `suppress(StandardError)`. Allowing `suppress <block>` does not make code difficult-to-debug, only add a syntax sugar to `suppress(StandardError)`.

File: activesupport/lib/active_support/core_ext/kernel/reporting.rb [hunk 1]
```diff
@@ -1,6 +1,9 @@
 module Kernel
   module_function
 
+  DEFAULT_SUPPRESS_EXCEPTIONS = [StandardError].freeze
```
Comment (rafaelfranca) [inline activesupport/lib/active_support/core_ext/kernel/reporting.rb]: No need to `freeze` the constant.

Comment (ixti) [inline activesupport/lib/active_support/core_ext/kernel/reporting.rb]: Idea was to disallow mutation of that list.
Although now I believe there's no need in that constant at all:

``` ruby
def suppress(*exceptions_list)
  exceptions_list = StandardError if exceptions_list.empty?
  # ...
end
```

**Comment by ixti** [comment]
@nateberkopec I agree with you at some degree and in fact I dislike ` tada rescue nil` nils either. But seeing `suppress(StandardError) { ... }` makes me a bit more sad :D

**Comment by ixti** [comment]
Sorry, accidentally pressed "Update branch" so it merged master into this PR.

**Comment by ixti** [comment]
Thanks!

**Comment by rafaelfranca** [comment]
@matthewd pointed that `suppress` without a argument is confusing since you don't know what it is suppressing and can be confused with `ActiveRecord::Base#suppress` so I reverted this change in 661ed30. Thank you for the pull request.

**Pull request merged**